api/src/oscc.c

```diff
@@ -492,110 +615,366 @@ oscc_result_t oscc_async_enable( int socket )
+    }
+    else
     {
         strncpy( ifr.ifr_name, can_channel, IFNAMSIZ );
 
-        ret = ioctl( sock, SIOCGIFINDEX, &ifr );
+        valid = ioctl( sock, SIOCGIFINDEX, &ifr );
 
-        if ( ret < 0 )
+        if ( valid < 0 )
         {
-            printf( "Finding CAN index failed: %s\n", strerror(errno) );
-
-            result = OSCC_ERROR;
+            perror( "Finding CAN index failed:" );
         }
     }
 
+    //If a timeout has been specified set one here since it should be set before
+    //the bind call
+    if( valid >= 0 && tv != NULL )
+    {
+        valid = setsockopt( sock,
+                            SOL_SOCKET,
+                            SO_RCVTIMEO,
+                            tv,
+                            sizeof( struct timeval ) );
+
+        if ( valid < 0 )
+        {
+            perror( "Setting timeout failed:" );
+        }
+    }
 
-    if ( result == OSCC_OK )
+    if ( valid >= 0 )
     {
         struct sockaddr_can can_address;
 
-        memset( &can_address, 0, sizeof(can_address) );
+        memset( &can_address, 0, sizeof( can_address ) );
         can_address.can_family = AF_CAN;
         can_address.can_ifindex = ifr.ifr_ifindex;
 
-        ret = bind(
-            sock,
-            (struct sockaddr *) &can_address,
-            sizeof(can_address) );
+        valid = bind( sock,
+                      (struct sockaddr *) &can_address,
+                      sizeof( can_address ) );
 
-        if ( ret < 0 )
+        if ( valid < 0 )
         {
-            printf( "Socket binding failed: %s\n", strerror(errno) );
-
-            result = OSCC_ERROR;
+            perror( "Socket binding failed:" );
         }
     }
 
+    // Clean up resources and close the connection if it's invalid.
+    if( valid < 0 )
+    {
+      close( sock );
+      sock = UNINITIALIZED_SOCKET;
+    }
 
-    if ( result == OSCC_OK )
+    return sock;
+}
+
+
+can_contains_s can_detection( const char *can_channel )
+{
+    struct timeval timeout;
+    timeout.tv_sec = 0;
+    timeout.tv_usec = CAN_MESSAGE_TIMEOUT;
+
+    int sock = init_can_socket( can_channel, &timeout );
+
+    vehicle_can_desc_s vehicle_detection =
     {
-        ret = oscc_async_enable( sock );
+        .has_steering_angle = false,
+        .has_brake_pressure = false,
+        .has_wheel_speed = false
+    };
 
-        if ( ret != OSCC_OK )
-        {
-            printf( "Enabling asynchronous socket I/O failed\n" );
+    oscc_can_desc_s oscc_detection =
+    {
+        .has_accel_report = false,
+        .has_steer_report = false,
+        .has_brake_report = false
+    };
 
-            result = OSCC_ERROR;
+    int i = 0;
+
+    for( i = 0; i < MAX_CAN_IDS; i++ )
+    {
+        struct can_frame rx_frame;
+        memset( &rx_frame, 0, sizeof( rx_frame ) );
+        int recv_bytes = 0;
+
+        recv_bytes = read( sock, &rx_frame, sizeof( rx_frame ) );
+
+        switch (recv_bytes) {
+          case CAN_MTU:
```

Comment:
most standards require a default case and a break for anything that isn't clearly falling through to another `case` (including the last or default case.)

Reply:
I just removed that per @adrienthebo's comment a few commits ago. Should I put this back in?

```
default:
    continue;
```